core/services/fluxmonitor/flux_monitor.go

```diff
@@ -878,3 +884,44 @@ func OutsideDeviation(curAnswer, nextAnswer decimal.Decimal, threshold float64)
 	logger.Infow("Deviation threshold met", loggerFields...)
 	return true
 }
+
+const pathologicalSkew = 30 * time.Second
+
+func MakeNewRoundIdleTimer(log *contracts.LogNewRound, idleThreshold models.Duration, clock Clock) <-chan time.Time {
+	timeNow := clock.Now()
+	var roundStarted time.Time
+	if log.StartedAt.IsInt64() {
+		roundStarted = time.Unix(log.StartedAt.Int64(), 0)
+	} else {
+		logger.Errorf("value for log.StartedAt %s would overflow int64, ignoring system clock and setting idle timer to max duration of %s", log.StartedAt.String(), idleThreshold.Duration())
+		return clock.After(idleThreshold.Duration())
+	}
+	if roundStarted.After(timeNow) {
+		logger.Warnf("round started time of %s is later than current system time of %s, ignoring system clock and setting idle timer to max duration of %s. This machine probably has a slow clock", roundStarted.String(), timeNow.String(), idleThreshold.Duration().String())
+		return clock.After(idleThreshold.Duration())
+	}
+	if timeNow.Sub(roundStarted) > pathologicalSkew {
+		logger.Warnf("current system time %s is much later than block timestamp of %s. Either you are processing an old round, or this machine has a fast clock", timeNow.String(), roundStarted.String())
+	}
+	// duration from now until idle threshold = log timestamp + idle threshold - current time
+	timeNow = clock.Now()
+	durationUntilIdleThreshold := roundStarted.Add(idleThreshold.Duration()).Sub(timeNow)
+	if durationUntilIdleThreshold < 0 {
+		logger.Warnf("idle threshold already passed, current time is %s and idle timer expired at %s", timeNow, roundStarted.Add(idleThreshold.Duration()).String())
+		expired := make(chan time.Time, 1)
+		expired <- roundStarted.Add(idleThreshold.Duration())
+		return expired
```

Comment:
Github lost my comment when @spooktheducks posted his as I was reviewing 😠  </rant>

I think these three lines could be removed since clock.After correctly handles negative durations.

Reply:
Are you sure? My testing does not indicate this.

Reply:
```go
package main

import (
	"fmt"
	"time"
)

func main() {
	fmt.Println("time at start", time.Now())
	select {
	case t:= <-time.After(-10 * time.Second):
		fmt.Println("timed out", t)
	}
}
```
prints (almost) the same time twice.